cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DistributedSchedulerService.java

```diff
@@ -78,6 +82,7 @@ protected void doStop() {
   protected void startUp() throws Exception {
     LOG.info("Starting scheduler.");
     serviceDelegate.startAndWait();
+    countDownLatch.await();
```

Comment:
My question is, does it react to stop correctly?? AFAIK, blocking in the `startUp` would make the service not able to react to `stop`, meaning you'll have a non-daemon running, blocking the termination of the master process.